extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/client/HTTPAppFactoryClient.java

```diff
@@ -0,0 +1,224 @@
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Handles all type of HTTP/HTTPS requests and responses from DevStudio to AppFactory/ AppCloud server
+ */
+public class HTTPAppFactoryClient {
+    private static final Logger log = LoggerFactory.getLogger(HTTPAppFactoryClient.class);
+
+    private static HttpClient client;
+    private static String cookie;
+
+    /**
+     * Creates an HTTP Entity using given request parameters
+     *
+     * @param serverURL          AppFactory/ AppCloud server URL to connect with for login request
+     * @param loginRequestParams Map that contains login request params as key value pairs
+     * @return Populated AppFactoryLoginResponse object
+     * @throws org.wso2.developerstudio.codenvy.ext.appfactory.server.exception.AppFactoryServerException
+     */
+    public static AppFactoryLoginResponse doLogin(String serverURL, Map<String, String> loginRequestParams)
+            throws AppFactoryServerException {
+        AppFactoryLoginResponse loginResponse = new AppFactoryLoginResponse();
+        String successLogin;
+
+        //Creating an http client and wrap it
+        client = new DefaultHttpClient();
+        client = wrapClient(client, serverURL);
+
+        //Creating a http post request by giving server URL
+        HttpPost httpLoginRequest = new HttpPost(serverURL);
+        HttpResponse httpLoginResponse;
+
+        try {
+            httpLoginRequest.setEntity(createEntityFromRequestParams(loginRequestParams));
+        } catch (UnsupportedEncodingException e) {
+            log.error("Error in encoding login request parameters to an entity, Parameters : " + loginRequestParams, e);
+            throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+        }
+
+        //Sending HTTP post request to AppFactory/ AppCloud
+        try {
+            httpLoginResponse = client.execute(httpLoginRequest);
+        } catch (IOException e) {
+            log.error("Error while sending HTTP Post request to server, Server URL : " + serverURL
+                    + ", HTTP Request : " + httpLoginRequest, e);
+            throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+        }
+
+        if (200 == httpLoginResponse.getStatusLine().getStatusCode()) {    //If response status is 200(OK)
+            //Extracting cookie value from response
+            cookie = httpLoginResponse.getFirstHeader("Set-Cookie").getValue().split(";")[0];
+
+            HttpEntity loginResponseEntity = httpLoginResponse.getEntity();
+            BufferedReader bufferedReader;
+            StringBuilder responseBuilder = new StringBuilder();
+            String responseLine;
+
+            try {
+                bufferedReader = new BufferedReader(new InputStreamReader(loginResponseEntity.getContent()));
```

Comment:
bufferedReader instance isn't closed after all the logic is invoked upon it. This can lead to memory leaks. Please fix.